### PR TITLE
Implements ParentNode#replaceChildren

### LINKED ISSUE #26240: Implement ParentNode#replaceChildren
Spec: whatwg/dom#851
Tests: web-platform-tests/wpt#22524

### PR DESCRIPTION
**Pull request by ghostd**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #26240

- [X] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

**Comment by jdm** [comment]
This will need to be rebased against a recent upstream revision. I recommend running `git reset --hard [revision]`, then `git cherry-pick 59059b392c283ef8c37234285090569d02478f1c`.

**Comment by ghostd** [comment]
Seems better now :)

**Comment by ghostd** [comment]
I'll look at the failing tests

File: components/script/dom/node.rs [hunk 1]
```diff
@@ -934,6 +934,18 @@ impl Node {
         self.AppendChild(&node).map(|_| ())
     }
 
+    // https://dom.spec.whatwg.org/#dom-parentnode-append
```
Comment (CYBAI) [inline components/script/dom/node.rs:937]:
```suggestion
    // https://dom.spec.whatwg.org/#dom-parentnode-replacechildren
```

Found wrong link though 👀

**Comment by CYBAI** [comment]
The failure is #25513 but maybe we can fix the link and approve this PR again 👀?

File: components/script/dom/node.rs (hunk 1 above)
Comment (ghostd) [reply components/script/dom/node.rs]: My bad :-/
Fixed and rebased.

Comment (CYBAI) [reply components/script/dom/node.rs]: Thank you so much ❤️

**Comment by CYBAI** [comment]
@bors-servo try

(I can't approve but I can try to trigger a `try` to see all tests pass and then the next r+ will merge this PR directly!)

**Comment by CYBAI** [comment]
@bors-servo retry

- #25513

**Pull request merged**